protocol/protocol.md

```diff
@@ -175,111 +308,424 @@ List directory _path_ and return mtime, size, type, etag and file id.
-              <d:prop>
-                <oc:id>00000287ocee55caa1e4</oc:id>
-                <d:getetag>"5391c8e1e4cf0"</d:getetag>
-                <d:getlastmodified>Fri, 06 Jun 2014 13:57:53 GMT</d:getlastmodified>
-                <d:resourcetype><d:collection/></d:resourcetype>
-              </d:prop>
-              <d:status>HTTP/1.1 200 OK</d:status>
-            </d:propstat>
-            <d:propstat>
-              <d:prop>
-                <d:getcontentlength/>
-              </d:prop>
-              <d:status>HTTP/1.1 404 Not Found</d:status>
-            </d:propstat>
-          </d:response>
-          <d:response>
-            <d:href>/remote.php/webdav/New%20folder/file.txt</d:href>
-            <d:propstat>
-              <d:prop>
-                <oc:id>00000292ocee55caa1e4</oc:id>
-                <d:getetag>"5391c8e19e24f"</d:getetag>
-                <d:getlastmodified>Fri, 06 Jun 2014 13:57:53 GMT</d:getlastmodified>
-                <d:getcontentlength>55</d:getcontentlength>
-                <d:resourcetype/>
-              </d:prop>
-              <d:status>HTTP/1.1 200 OK</d:status>
-            </d:propstat>
-          </d:response>
-        </d:multistatus>  
-
-
-### Create directory
+### Server Directory Creation
+
+Client sends: WebDAV request `MKCOL path`
+
+Expected success result: HTTP Reply Code 201
+
+Example:
+
+    MKCOL http://mycomputer/oc/remote.php/webdav/documentation
+
+Request Custom Header: None
+
+Response Custom Header:
+
+`OC-FileId`: The fileId of the new directory.
+
+Specific error code handling:
 
 ### Move file or directory
 
-### Remove file or directory
+Client sends: HTTP request `MOVE path` with destination set.
+
+Expected success result: HTTP Reply Code 201
+
+Example:
+
+    MOVE http://mycomputer/oc/remote.php/webdav/welcome.txt
+    Destination: /oc/remote.php/webdav/welcome_kf.txt
+
+Request Custom Header: None
+
+Response Custom Header:
+
+`OC-FileId`: The fileId of the new directory.
+
+Specific error code handling: None
 
-    
 ## HTTP
 
-### Download file
-### Upload file
-### Upload files (chunked)
+### Server Directory Removal
+
+Client sends: HTTP request `DELETE path`
+
+Expected success result: HTTP Reply Code 204
+
+Example:
+
+    DELETE http://mycomputer/oc/remote.php/webdav/documentation
+
+Request Custom Header: None
+
+Response Custom Header:
+
+`OC-FileId`: The fileId of the new directory.
+
+Specific error code handling:
+
+* `403` - removed local share directory without permission
+      The client will try to recover the file from server in a subsequent
+      sync run.
+* `404` - Not considered an error, because directory is already gone.
+
+### Server File Removal:
+
+Client sends: HTTP request `DELETE file`
+
+Expected success result: HTTP Reply Code 204
+
+Example:
+
+    DELETE http://mycomputer/oc/remote.php/webdav/documentation/mybook.txt
+
+Request Custom Header: None
+
+Response Custom Header:
+
+`OC-FileId`: The fileId of the new directory.
```

Comment:
Well, that is a mistake, of course the response to DELETE does not contain any FileId header. Sorry.